util/timer.h

```diff
@@ -121,17 +123,20 @@ class Timer {
   bool Shutdown() {
     {
       InstrumentedMutexLock l(&mutex_);
+      shutting_down_ = true;
```

Comment:
Perhaps I do not understand some of the cases enough but won't this cause the Start task to wait forever if running_ = false?  Right below here it returns false "if (!running_)", without resetting shutdown or signaling any conditions.

Reply:
Nice catch. Thanks @mrambacher